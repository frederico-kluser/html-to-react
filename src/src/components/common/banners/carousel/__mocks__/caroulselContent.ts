export const bannerCarousel = {
  bannerId: 2,
  type: 2,
  medias: [
    {
      imageUrl: 'https://m.media-amazon.com/images/I/61CiqVTRBEL._SX1500_.jpg',
      videoUrl: null,
      text: null,
      content: null,
      href: 'https://www.amazon.com/b/?_encoding=UTF8&node=23508887011&pf_rd_r=NGQQXRSZEX2BTSSXFMSE&pf_rd_p=b3367632-3703-47fc-9e4a-042ab9bd1208',
      order: 3
    },
    {
      imageUrl: 'https://m.media-amazon.com/images/I/61TD5JLGhIL._SX1500_.jpg',
      href: 'https://www.amazon.com/b/?_encoding=UTF8&node=16225007011&pf_rd_r=NGQQXRSZEX2BTSSXFMSE&pf_rd_p=da67e2af-3930-4aa9-9ced-7c541705ee2b',
      videoUrl: null,
      text: null,
      content: null,
      order: 1
    }
  ],
  order: 1
};

export const bannerCarouselWithoutHref = {
  bannerId: 2,
  type: 2,
  medias: [
    {
      imageUrl: 'https://m.media-amazon.com/images/I/61CiqVTRBEL._SX1500_.jpg',
      videoUrl: null,
      text: null,
      content: null,
      href: null,
      order: 3
    }
  ],
  order: 1
};

export const bannerCarouselWithoutImage = {
  bannerId: 2,
  type: 2,
  medias: [
    {
      imageUrl: null,
      videoUrl: null,
      text: null,
      content: null,
      href: 'https://www.amazon.com/b/?_encoding=UTF8&node=23508887011&pf_rd_r=NGQQXRSZEX2BTSSXFMSE&pf_rd_p=b3367632-3703-47fc-9e4a-042ab9bd1208',
      order: 1
    }
  ],
  order: 1
};
