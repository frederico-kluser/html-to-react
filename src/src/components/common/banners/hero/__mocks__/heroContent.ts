export const bannerHero = {
  bannerId: 1,
  type: 1,
  medias: [
    {
      imageUrl: 'https://m.media-amazon.com/images/I/61CiqVTRBEL._SX1500_.jpg',
      href: 'https://www.amazon.com/b/?_encoding=UTF8&node=23508887011&pf_rd_r=NMDG71DKN5XRWHGVWCQ6&pf_rd_p=b3367632-3703-47fc-9e4a-042ab9bd1208',
      videoUrl: null,
      text: null,
      content: null,
      order: 0
    }
  ],
  order: 0
};
