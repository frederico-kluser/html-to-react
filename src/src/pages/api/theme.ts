import { connectToDatabase } from '../../database/connect';
import type { NextApiRequest, NextApiResponse } from 'next';

export const getTheme = async () => {
  const { db } = await connectToDatabase();
  return await db.collection('themes').findOne({ name: 'Light' });
};

const handler = async (req: NextApiRequest, res: NextApiResponse) => {
  const theme = await getTheme();
  res.status(200).json(theme);
};

export default handler;
