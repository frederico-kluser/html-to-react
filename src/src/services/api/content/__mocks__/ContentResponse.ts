export const ContentResponse = {
  data: {
    pageId: 1,
    pageName: 'Home',
    title: null,
    breadcrumb: null,
    sections: [
      {
        content: '<h1 data-testid="heading-level-1-test">Teste</h1>',
        banners: [
          {
            bannerId: 2,
            type: 2,
            medias: [
              {
                imageUrl:
                  'https://m.media-amazon.com/images/I/61CiqVTRBEL._SX1500_.jpg',
                videoUrl: null,
                text: null,
                content: null,
                href: 'https://www.amazon.com/b/?_encoding=UTF8&node=23508887011&pf_rd_r=NGQQXRSZEX2BTSSXFMSE&pf_rd_p=b3367632-3703-47fc-9e4a-042ab9bd1208',
                order: 3
              },
              {
                imageUrl:
                  'https://m.media-amazon.com/images/I/61TD5JLGhIL._SX1500_.jpg',
                href: 'https://www.amazon.com/b/?_encoding=UTF8&node=16225007011&pf_rd_r=NGQQXRSZEX2BTSSXFMSE&pf_rd_p=da67e2af-3930-4aa9-9ced-7c541705ee2b',
                videoUrl: null,
                text: null,
                content: null,
                order: 1
              },
              {
                imageUrl:
                  'https://m.media-amazon.com/images/I/61jovjd+f9L._SX1500_.jpg',
                href: 'https://www.amazon.com/b/?_encoding=UTF8&node=16225015011&pf_rd_r=NGQQXRSZEX2BTSSXFMSE&pf_rd_p=466f6c10-383b-44c2-a942-4b0b7f6295c9',
                videoUrl: null,
                text: null,
                content: null,
                order: 2
              }
            ],
            order: 1
          },
          {
            bannerId: 1,
            type: 1,
            medias: [
              {
                imageUrl:
                  'https://m.media-amazon.com/images/I/61CiqVTRBEL._SX1500_.jpg',
                href: 'https://www.amazon.com/b/?_encoding=UTF8&node=23508887011&pf_rd_r=NMDG71DKN5XRWHGVWCQ6&pf_rd_p=b3367632-3703-47fc-9e4a-042ab9bd1208',
                videoUrl: null,
                text: null,
                content: null,
                order: 0
              }
            ],
            order: 0
          }
        ],
        shelf: [
          // {
          //   shelfId: 2,
          //   type: 2,
          //   collection: {
          //     collectionId: 100,
          //     collectionType: {
          //       id: 2,
          //       description: 'Mais Vendidos da Categoria'
          //     },
          //     skus: [1040, 5050, 474554, 654654, 8787]
          //   }
          // },
          // {
          //   shelfId: 3,
          //   type: 1,
          //   collection: {
          //     collectionId: 100,
          //     collectionType: {
          //       id: 3,
          //       description: 'Todos os produtos da categoria'
          //     },
          //     skus: [100, 5456, 231, 321, 564, 9875456]
          //   }
          // },
          // {
          //   shelfId: 4,
          //   type: 99999999,
          //   collection: {
          //     collectionId: 100,
          //     collectionType: {
          //       id: 3,
          //       description: 'Todos os produtos da categoria'
          //     },
          //     skus: [100, 5456, 231, 321, 564, 9875456]
          //   }
          // }
        ]
      },
      {
        content: null,
        banners: null,
        shelf: null
      }
    ],
    navigation: [
      {
        id: 'moveis',
        text: 'Móveis',
        url: 'https://m.casasbahia.com.br/moveis',
        children: [
          {
            text: 'Cama',
            url: 'https://m.casasbahia.com.br/cama/moveis',
            children: null
          }
        ]
      },
      {
        id: 'telefones-e-celulares',
        text: 'Telefones & Celulares',
        url: 'https//m.casasbahia.com.br/telefones-e-celulares',
        children: []
      }
    ],
    links: [
      {
        id: 'after-sale',
        text: 'Atendimento',
        url: 'https://meusite.com.br/atendimento'
      },
      {
        id: 'cart',
        text: 'Meu carrinho',
        url: 'https://meusite.com.br/cart'
      },
      {
        id: 'my-account',
        text: 'Minha Conta',
        url: 'https://meusite.com.br/account'
      },
      {
        id: 'wedding-list',
        text: 'Lista de casamento',
        url: 'https://meusite.com.br/wedding-list'
      },
      {
        id: 'orders',
        text: 'Meus Pedidos',
        url: 'https://account.meusite.com.br/orders'
      }
    ],
    'company-contact-data': {
      storeName: 'Giorgio Armany',
      street: 'Av. Magalhães de Castro, 12000',
      zipCode: '09362-325',
      city: 'São Paulo',
      state: 'SP'
    },
    products: [
      1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21,
      23
    ],
    filters: {
      prices: [
        {
          priceMin: 1000,
          priceMax: 3000,
          skusCount: 1589
        },
        {
          priceMin: 3000,
          priceMax: 6500,
          skusCount: 2000
        }
      ],
      brands: [
        {
          brandId: 1,
          name: 'Apple',
          skusCount: 30
        }
      ],
      categories: [
        {
          categoryId: 100,
          name: 'Acessórios e Inovações',
          parent: {
            categoryId: 1,
            name: 'Celulares e Telefones'
          },
          children: [],
          skusCount: 3737
        },
        {
          categoryId: 100,
          name: 'Smartphones',
          parent: {
            categoryId: 1,
            name: 'Celulares e Telefones'
          },
          children: [],
          skusCount: 1157
        }
      ],
      discounts: [
        {
          discountMin: 1000,
          discountMax: 3000,
          skusCount: 10
        },
        {
          discountMin: 2000,
          discountMax: 6500,
          skusCount: 50
        }
      ],
      attributes: [
        {
          attributeId: 12,
          attributeName: 'Tamanho da tela',
          attributeValues: [
            {
              attributeValueId: 121,
              attributeValue: 'Acima de 6',
              skusCount: 58
            },
            {
              attributeValueId: 122,
              attributeValue: 'De 5.1 a 5.9',
              skusCount: 16
            }
          ]
        },
        {
          attributeId: 1,
          attributeName: 'Cor',
          attributeValues: [
            {
              attributeValueId: 300,
              attributeValue: 'Branco',
              skusCount: 47
            },
            {
              attributeValueId: 301,
              attributeValue: 'Preto',
              skusCount: 40
            }
          ]
        }
      ]
    },
    recordsReturned: 100,
    totalRecords: 15000
  }
};

export const heroBanner = {
  bannerId: 1,
  type: 1,
  medias: [
    {
      imageUrl:
        'https://images.unsplash.com/photo-1572584642822-6f8de0243c93?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=1200&h=600&q=50',
      href: 'https://www.amazon.com/b/?_encoding=UTF8&node=23508887011&pf_rd_r=NMDG71DKN5XRWHGVWCQ6&pf_rd_p=b3367632-3703-47fc-9e4a-042ab9bd1208',
      videoUrl: null,
      text: null,
      content: null,
      order: 0
    }
  ],
  order: 0
};

export const carouselContent = {
  bannerId: 2,
  type: 2,
  medias: [
    {
      imageUrl: 'https://m.media-amazon.com/images/I/61CiqVTRBEL._SX1500_.jpg',
      videoUrl: null,
      text: null,
      content: null,
      href: 'https://www.amazon.com/b/?_encoding=UTF8&node=23508887011&pf_rd_r=NGQQXRSZEX2BTSSXFMSE&pf_rd_p=b3367632-3703-47fc-9e4a-042ab9bd1208',
      order: 0
    },
    {
      imageUrl: 'https://m.media-amazon.com/images/I/61TD5JLGhIL._SX1500_.jpg',
      href: 'https://www.amazon.com/b/?_encoding=UTF8&node=16225007011&pf_rd_r=NGQQXRSZEX2BTSSXFMSE&pf_rd_p=da67e2af-3930-4aa9-9ced-7c541705ee2b',
      videoUrl: null,
      text: null,
      content: null,
      order: 1
    },
    {
      imageUrl: 'https://m.media-amazon.com/images/I/61jovjd+f9L._SX1500_.jpg',
      href: 'https://www.amazon.com/b/?_encoding=UTF8&node=16225015011&pf_rd_r=NGQQXRSZEX2BTSSXFMSE&pf_rd_p=466f6c10-383b-44c2-a942-4b0b7f6295c9',
      videoUrl: null,
      text: null,
      content: null,
      order: 1
    }
  ],
  order: 1
};

export const carouselContentNoImage = {
  bannerId: 2,
  type: 2,
  medias: [
    {
      imageUrl: null,
      videoUrl: null,
      text: null,
      content: null,
      href: 'https://www.amazon.com/b/?_encoding=UTF8&node=23508887011&pf_rd_r=NGQQXRSZEX2BTSSXFMSE&pf_rd_p=b3367632-3703-47fc-9e4a-042ab9bd1208',
      order: 0
    }
  ],
  order: 1
};

export const carouselContentWithoutHref = {
  bannerId: 2,
  type: 2,
  medias: [
    {
      imageUrl: 'https://m.media-amazon.com/images/I/61jovjd+f9L._SX1500_.jpg',
      videoUrl: null,
      text: null,
      content: null,
      href: null,
      order: 0
    }
  ],
  order: 1
};
