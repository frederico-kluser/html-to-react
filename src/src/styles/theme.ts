export const theme = {
  id: 'T1',
  name: 'Light',
  layout: {
    type: 'fixed',
    maxWidth: 1200,
    widthUnit: 'px'
  },
  colors: {
    primary: {
      main: '#0C1446',
      common: {
        black: '#000',
        white: '#fff',
        gray: '#616161',
        bgColor: '#af9995'
      }
    }
  },
  breakpoints: {
    size: {
      xs: 0,
      sm: 600,
      md: 960,
      lg: 1280,
      xl: 1920
    },
    unit: 'px'
  }
};
