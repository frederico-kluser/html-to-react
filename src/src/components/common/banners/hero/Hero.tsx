/* eslint-disable @next/next/no-img-element */
import React from 'react';
import { IBanner } from '../../../../models/content';
import * as S from './Hero.styles';
import HeroImageContent from './HeroImageContent';

type HeroProps = IBanner & {
  /**
   * renders observing the max width defined in the theme or
   * uses all available horizontal space
   */
  layoutType?: 'fixed' | 'fluid';
};

const Hero = (props: HeroProps) => {
  const { medias, bannerId, layoutType = 'fixed' } = props;
  const [media] = medias;
  const { imageUrl, href } = media;

  if (!imageUrl) {
    return (
      <p data-testid={`image-url-not-provided-${bannerId}`}>
        Only banners with imageUrl attribute are supported: {bannerId}
      </p>
    );
  }

  const Container =
    layoutType === 'fixed' ? S.ContainerFixed : S.ContainerFluid;

  const content = (
    <HeroImageContent id={bannerId} imageUrl={imageUrl} href={href} />
  );

  return (
    <Container data-testid={`hero-banner-${bannerId}`}>{content}</Container>
  );
};

export default Hero;
