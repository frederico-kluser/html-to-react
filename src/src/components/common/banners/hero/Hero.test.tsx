import { render, screen } from '@testing-library/react';
import { ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../../../styles/theme';
import Hero from '.';
// TODO Import this Mock
import { bannerHero } from './__mocks__/heroContent';
import HeroContent from './HeroImageContent';

const renderComponent = (children: ReactNode) =>
  render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);

describe('<Hero />', () => {
  it('Should render the Hero', () => {
    const { container } = renderComponent(<Hero {...bannerHero} />);

    const testId = `hero-banner-${bannerHero.bannerId}`;
    expect(screen.getByTestId(testId)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('Should render a message when no Image URL is provided', () => {
    const bannerWithNoImage = { ...bannerHero };
    bannerWithNoImage.medias[0].imageUrl = '';
    const { container } = renderComponent(<Hero {...bannerWithNoImage} />);

    const testId = `image-url-not-provided-${bannerHero.bannerId}`;
    expect(screen.getByTestId(testId)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('Should render a image whithout link when no Href is provided', () => {
    const { medias } = bannerHero;
    const [media] = medias;

    const { container } = renderComponent(
      <HeroContent id={1} href={null} imageUrl={media.imageUrl} />
    );

    const testId = `hero-content-${bannerHero.bannerId}`;
    expect(screen.getByTestId(testId)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
