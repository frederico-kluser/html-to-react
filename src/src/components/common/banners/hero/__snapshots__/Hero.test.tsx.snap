// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> Should render a image whithout link when no Href is provided 1`] = `
<img
  alt="banner 1"
  data-testid="hero-content-1"
  src=""
/>
`;

exports[`<Hero /> Should render a message when no Image URL is provided 1`] = `
<p
  data-testid="image-url-not-provided-1"
>
  Only banners with imageUrl attribute are supported: 
  1
</p>
`;

exports[`<Hero /> Should render the Hero 1`] = `
.c0 {
  margin: auto;
  max-width: 1200px;
}

.c0 img {
  max-width: 100%;
}

@media (max-width:960px) {
  .c0 {
    max-width: 100%;
  }
}

<figure
  class="c0"
  data-testid="hero-banner-1"
>
  <a
    href="https://www.amazon.com/b/?_encoding=UTF8&node=23508887011&pf_rd_r=NMDG71DKN5XRWHGVWCQ6&pf_rd_p=b3367632-3703-47fc-9e4a-042ab9bd1208"
  >
    <img
      alt="banner 1"
      data-testid="hero-content-1"
      src="https://m.media-amazon.com/images/I/61CiqVTRBEL._SX1500_.jpg"
    />
  </a>
</figure>
`;
